seshdash/utils/send_slack.py

```diff
@@ -0,0 +1,38 @@
+""" 
+For sending messages to Slack
+"""
+# Python libs
+import logging
+
+from django.conf import settings
+
+from slacker import Slacker
+
+
+class Slack():
+    
+    def __init__(self, auth_token):
+        self.slack_client = Slacker(auth_token)
+        
+    def send_message_to_channel(self, channel, message):
+        """ Sends a message to a slack channel """
+        if channel[0] == '#':
+            pass
+        else:
+            channel = '#' + channel
+     
+              
+        try:
+            # Posting the message
+            self.slack_client.chat.post_message(channel, message)
+            return True
+        except Exception, e: 
```

Comment:
Here you need to actually check which exception is being thrown.
So if there is a specific slack exception that says channel does not exist you should catch that and do the necessary steps to create the channel. `except Exception,e` will create a channel on all exceptions regardless of the exception.